api/graphql/types/reservation_unit/serializers.py

```diff
@@ -206,34 +202,52 @@ def validate_for_publish(self, data: dict[str, Any]) -> None:
             msg = "minPersons can't be more than maxPersons"
             raise ValidationError(msg, code=error_codes.RESERVATION_UNIT_MIN_PERSONS_GREATER_THAN_MAX_PERSONS)
 
-    def validate_created_pricings(self, data: dict[str, Any]) -> dict[str, Any]:
+    def _validate_pricings(self, data: dict[str, Any]) -> None:
+        """
+        When reservation unit is draft, pricings are not required,
+        but if they are given, they must be valid.
+        """
         is_draft = self.get_or_default("is_draft", data)
-        ReservationUnitPricingHelper.check_pricing_required(is_draft, data)
-        ReservationUnitPricingHelper.check_pricing_dates(data)
-        ReservationUnitPricingHelper.check_pricing_counts(is_draft, data)
-        data["pricings"] = ReservationUnitPricingHelper.calculate_vat_prices(data)
-        return data
+        pricings = data.get("pricings")
+
+        # Pricings are optional when reservation unit is draft
+        if is_draft and pricings is None:
+            return
 
-    def validate_updated_pricings(self, data: dict[str, Any]) -> dict[str, Any]:
-        current_active_pricing = ReservationUnitPricingHelper.get_active_price(self.instance)
-        current_future_pricing = ReservationUnitPricingHelper.get_future_price(self.instance)
+        # If the pricings were not given, but the reservation unit has some, skip validation and assume they are valid
+        if pricings is None and self.instance and self.instance.pricings.exists():
+            return
 
-        for pricing in data.get("pricings", []):
-            if (
-                ReservationUnitPricingHelper.is_active(pricing)
-                and current_active_pricing
-                and current_active_pricing.pk != pricing.get("pk", 0)
-            ):
-                raise GraphQLError("ACTIVE pricing is already defined. Only one ACTIVE pricing is allowed")
+        if pricings is None:
+            msg = "Pricings are required for non-draft reservation units."
+            raise ValidationError(msg, code=error_codes.RESERVATION_UNIT_PRICINGS_MISSING)
 
-            if (
-                ReservationUnitPricingHelper.is_future(pricing)
-                and current_future_pricing
-                and current_future_pricing.pk != pricing.get("pk", 0)
-            ):
-                raise GraphQLError("FUTURE pricing is already defined. Only one FUTURE pricing is allowed")
+        # At least one given pricing must be currently active (begin date is today or earlier)
+        today = local_date()
+        has_active_pricing = any(p.get("begins") <= today for p in pricings)
```

Comment:
I'm guessing pricings always have a begins date? Otherwise this will result in a "None <= date" comparison, which will error.

Reply:
Yep `begins` is required. Added a test for this.